### ISSUE TITLE
Feature request: create file if not exists

### ISSUE THREAD

**Issue by aaronols** [robstoll/niok#23]
While using your library to work with `java.nio.file.Path` instances I found myself missing a function to "just make sure a file exists".
With plain Java, the next best fix would be to create some helper class that handles the problem in a generic manner.
But with Kotlin, this can be an extension function 😄 

My proposal is the addition of the following function to this library:

```kotlin
inline fun Path.createFileIfNotExists(vararg fileAttributes: FileAttribute<*>): Path =
    apply { try { createFile(fileAttributes) } catch (ignored: FileAlreadyExistsException) {} }
```

The longer (more readable form is the following:

```kotlin
inline fun Path.createFileIfNotExists(vararg fileAttributes: FileAttribute<*>): Path =
    this.also { path ->
        try {
            path.createFile(fileAttributes)
        } catch (ignored: FileAlreadyExistsException) {}
    }
```

The implementation uses another extension function provided by your library: Path.createFile().
It is of course possible to delegate directly to the java.nio.file.Files API which would change the semantics a bit and save an intermediate method call.

Also, the above proof of concept code is untested. Maybe inline functions have constraints that require a non-inline solution for this functionality.

I can provide a merge request if you are interested in this functionality.

**Comment by robstoll** [comment]
@aaronols sounds good to me, I would have implemented the same way (with `apply` but I would have put try and catch on separate lines). Please go ahead and create a pull request :)
I suggest you add a CreateFileIfNotExistsSpec

**Comment by robstoll** [comment]
@aaronols let me know in case you need help with the PR.

**Comment by robstoll** [comment]
implemented with 44fd2045 alongside of createDirectoryIfNotExists
